packages/hdwallet-native/src/cosmos.ts

```diff
@@ -34,10 +36,49 @@ export function MixinNativeCosmosWalletInfo<TBase extends core.Constructor>(Base
       // Only support one account for now (like portis).
       return undefined;
     }
+
+    cosmosDescribePath(path: BIP32Path): PathDescription {
+      let pathStr = addressNListToBIP32(path);
+      let unknown: PathDescription = {
+        verbose: pathStr,
+        coin: "Atom",
+        isKnown: false,
+      };
+
+      if (path.length != 5) {
+        return unknown;
+      }
+
+      if (path[0] != 0x80000000 + 44) {
+        return unknown;
+      }
+
+      if (path[1] != 0x80000000 + slip44ByCoin("Atom")) {
+        return unknown;
+      }
+
+      if ((path[2] & 0x80000000) >>> 0 !== 0x80000000) {
+        return unknown;
+      }
+
+      if (path[3] !== 0 || path[4] !== 0) {
+        return unknown;
+      }
+
+      let index = path[2] & 0x7fffffff;
+      return {
+        verbose: `Cosmos Account #${index}`,
+        accountIdx: index,
+        wholeAccount: true,
+        coin: "Atom",
+        isKnown: true,
+        isPrefork: false,
+      };
+    }
   };
 }
 
-export function MixinNativeCosmosWallet<TBase extends core.Constructor>(Base: TBase) {
+export function MixinNativeCosmosWallet<TBase extends core.Constructor<NativeHDWalletBase>>(Base: TBase) {
```

Comment:
CosmosWallet needs the `needsMnemonic` wrapper where applicable

Reply:
Can't believe I missed that. Thanks.